plugins/cilium-cni/cilium-cni.go

```diff
@@ -389,6 +389,26 @@ func cmdAdd(args *skel.CmdArgs) (err error) {
 
 	conf := *configResult.Status
 
+	podName := string(cniArgs.K8S_POD_NAMESPACE) + "/" + string(cniArgs.K8S_POD_NAME)
+	ipam, err = c.IPAMAllocate("", podName, true)
+	if err != nil {
+		err = fmt.Errorf("unable to allocate IP via local cilium agent: %s", err)
+		return
+	}
+
+	if ipam.Address == nil {
+		err = fmt.Errorf("Invalid IPAM response, missing addressing")
+		return
+	}
+
+	// release addresses on failure
+	defer func() {
+		if err != nil && ipam != nil && ipam.Address != nil {
+			releaseIP(c, ipam.Address.IPV4)
+			releaseIP(c, ipam.Address.IPV6)
+		}
+	}()
+
```

Comment:
should we move the below validation earlier as well ?

```
	if err = connector.SufficientAddressing(ipam.HostAddressing); err != nil {
		err = fmt.Errorf("IP allocation addressing in insufficient: %s", err)
		return
	}
```

Reply:
updated. running the test now :+1: